source/dem/force_chains_visualization.cc

```diff
@@ -0,0 +1,500 @@
+                        contact_info,
+                        tangential_relative_velocity,
+                        normal_relative_velocity_value,
+                        normal_unit_vector,
+                        normal_overlap,
+                        particle_one_properties,
+                        particle_two_properties,
+                        normal_force,
+                        tangential_force,
+                        particle_one_tangential_torque,
+                        particle_two_tangential_torque,
+                        rolling_resistance_torque);
+                    }
+
+                  if constexpr (contact_model ==
+                                Parameters::Lagrangian::
+                                  ParticleParticleContactForceModel::
+                                    hertz_mindlin_limit_force)
+                    {
+                      this->calculate_hertz_mindlin_limit_force_contact(
+                        contact_info,
+                        tangential_relative_velocity,
+                        normal_relative_velocity_value,
+                        normal_unit_vector,
+                        normal_overlap,
+                        particle_one_properties,
+                        particle_two_properties,
+                        normal_force,
+                        tangential_force,
+                        particle_one_tangential_torque,
+                        particle_two_tangential_torque,
+                        rolling_resistance_torque);
+                    }
+
+                  if constexpr (contact_model ==
+                                Parameters::Lagrangian::
+                                  ParticleParticleContactForceModel::
+                                    hertz_mindlin_limit_overlap)
+                    {
+                      this->calculate_hertz_mindlin_limit_overlap_contact(
+                        contact_info,
+                        tangential_relative_velocity,
+                        normal_relative_velocity_value,
+                        normal_unit_vector,
+                        normal_overlap,
+                        particle_one_properties,
+                        particle_two_properties,
+                        normal_force,
+                        tangential_force,
+                        particle_one_tangential_torque,
+                        particle_two_tangential_torque,
+                        rolling_resistance_torque);
+                    }
+
+                  vertices.push_back(particle_one_location);
+                  vertices.push_back(particle_two_location);
+                  force_normal.push_back(sqrt(normal_force.norm()));
+                }
+
+              else
+                {
+                  // if the adjacent pair is not in contact anymore, only the
+                  // tangential overlap is set to zero
+                  contact_info.tangential_overlap.clear();
+                }
```

Comment:
this is not important in this case because you don't have a tangential overlap .